gnovm/pkg/gnolang/alloc.go

```diff
@@ -133,6 +144,14 @@ func (alloc *Allocator) Allocate(size int64) {
 			}
 		}
 	}
+	if alloc.bytes > alloc.maxBytesUsed {
```

Comment:
maxBytesUsed == size?

Reply:
I think after the `GC` the `bytes` size can change (we have "dealloation")so it is why I do this check after `alloc.collect()`. if am not wrong @ltzmaxwell

Reply:
allocate does not receive negative values, and yes GC will decrease alloc.bytes.

Reply:
so the idea is to charge gas only monotonically increasing. if GC happened, the free space won't be double charged,  right?

Reply:
> allocate does not receive negative values, and yes GC will decrease alloc.bytes.
> 
> so the idea is to charge gas only for monotonically increasing part, if GC happened, the free space won't be double charged, right?

yes it is that